workorder-recheck.lua

```diff
@@ -3,18 +3,66 @@
 
 workorder-recheck
 =================
-Sets the status to ``Checking`` (from ``Active``) of the selected work order (in the ``j-m`` or ``u-m`` screens).
+Sets the status to ``Checking`` (from ``Active``) of the selected work order.
 This makes the manager reevaluate its conditions.
 ]====]
-local scr = dfhack.gui.getCurViewscreen()
-if df.viewscreen_jobmanagementst:is_instance(scr) then
-    local orders = df.global.world.manager_orders
-    local idx = scr.sel_idx
-    if idx < #orders then
-        orders[idx].status.active = false
+
+--@ module = true
+
+local gui = require('gui')
+local widgets = require('gui.widgets')
+local overlay = require('plugins.overlay')
+
+local function set_current_inactive()
+    local scrConditions = df.global.game.main_interface.info.work_orders.conditions
+    if scrConditions.open then
```

Comment:
this is the condition that would allow the overlay to be shown or the focus string to match below. I suppose it doesn't hurt terribly to check it again, but I don't think it's necessary.

Reply:
I added it in case something went terribly wrong. I guess the only possible scenario is if the overlay would somehow be active when it shouldn't...